crud/common/utils.lua

```diff
@@ -338,6 +357,49 @@ function utils.format_result(rows, space, field_names)
     return result
 end
 
+function utils.update_keys_fieldno(key_parts, start_fieldno)
+    local updated_key_parts = {}
+    start_fieldno = start_fieldno or 1
+    local current_fieldno = start_fieldno
+
+    for _, part in ipairs(key_parts) do
+        local updated_part = table.copy(part)
+        updated_part.fieldno = current_fieldno
+        current_fieldno = current_fieldno + 1
+        table.insert(updated_key_parts, updated_part)
+    end
+
+    return updated_key_parts
+end
+
+function utils.merge_comparison_fields(space_format, key_parts, field_names)
+    dev_checks('table', 'table', '?table')
+
+    if field_names == nil then
+        return nil
+    end
+
+    local merged_field_names = {}
+    local key_field_names = {}
+
+    for _, part in ipairs(key_parts) do
+        local field_name = space_format[part.fieldno].name
+        if not key_field_names[field_name] then
+            table.insert(merged_field_names, field_name)
+            key_field_names[field_name] = true
+        end
+    end
+
+    for _, field_name in ipairs(field_names) do
+        if not key_field_names[field_name] then
+            table.insert(merged_field_names, field_name)
+            key_field_names[field_name] = true
+        end
+    end
```

Comment:
I suggest to merge cmp_parts to field_names but not vice versa. Details I'll describe below - it helps to avoid some complex transformations. Finally you need to truncate "redundant" parts of tuple.